test/stac/stac-item.ts

```diff
@@ -328,4 +315,178 @@ describe('STAC item route', function () {
       });
     });
   });
+
+  describe('For a guest user who does not own the job', function () {
```

Comment:
It looks like most of the steps here are copied from another test. I'd just keep the one for a 409 when the job isn't complete and then a single 200 with JSON and get rid of all the other parts that are already tested elsewhere. Or if you want to just add a loop around the original test to test with different users.

Reply:
I didn't do the refactoring because I thought it was more readable this way. I will refactor the test as requested.